### PR TITLE
Split ESP32 ESP-IDF v3 and v4 travis builds

### PR DESCRIPTION
**Pull request by tve**
I'm trying to split the esp-idf v3 and v4 travis jobs to make it easier to see what is happening when there is a failure. The majority of the time is taken by the actual make process so the extra overhead of the double download seems minimal (and it's at travis' expense anyway).

### CODE REVIEW

**Comment by tve** [comment]
Looks like it now takes 3min for each of the esp-idf v3/v4 runs, vs. somewhat over 4min before. Not a huge time gain when run in parallel but makes it easier to find the problem when there is one.

**Comment by tve** [comment]
I forgot to mention that this PR also fixes the fact that the esp32 w/esp-idf-v4 firmware is being built with the wrong GCC 5.2.0 instead of the correct GCC 8.2.0.

File: .travis.yml [hunk 1]
```diff
@@ -234,28 +234,35 @@ jobs:
...
         - make ${MAKEOPTS} -C ports/windows CROSS_COMPILE=i686-w64-mingw32-
 
-    # esp32 port
+    # esp32 w/ESP-IDFv3 port
     - stage: test
-      env: NAME="esp32 port build"
+      env: NAME="esp32 ESP-IDFv3 port build"
       install:
         - sudo apt-get install python3-pip
         - sudo pip3 install 'pyparsing<2.4'
```
Comment (dpgeorge) [inline .travis.yml:242]: I think pyparsing is only needed for IDFv4, so both these pip lines can go.

**Comment by dpgeorge** [comment]
> I forgot to mention that this PR also fixes the fact that the esp32 w/esp-idf-v4 firmware is being built with the wrong GCC 5.2.0 instead of the correct GCC 8.2.0.

Ok, that's a good fix.

**Comment by dpgeorge** [comment]
Ahh, pyparsing is needed for IDF v3, it's just that it can be the latest version.

I've merged this in e880c8dfaac9d45fea7b086ac5e7d45e0e92e57e without the commit that removes pyparsing for IDF v3.

**Pull request closed without merging**